group/assignment5/AlphabetFromDictionary.cpp

```diff
@@ -1,120 +1,175 @@
 #include "AlphabetFromDictionary.hpp"
 
 AlphabetFromDictionary::AlphabetFromDictionary(const vector<string> &dictionary) : dictionary(dictionary) {
-    this->setDictionary(dictionary);
+	this->setDictionary(dictionary);
 }
 
 void AlphabetFromDictionary::setDictionary(const vector<string> &dictionary) {
-    this->dictionary = dictionary;
-    this->initializeClassVars();
-    this->findAlphabet();
+	this->dictionary = dictionary;
+	this->initializeClassVars();
+	this->findAlphabet();
 }
 
 vector<char> AlphabetFromDictionary::getAlphabet() {
-    return this->orderedAlphabet;
+	return this->orderedAlphabet;
 }
 
 bool AlphabetFromDictionary::isDictionaryConsistent() {
-    return !this->isGraphCyclic;
+	return !this->isGraphCyclic;
 }
 
 void AlphabetFromDictionary::initializeClassVars() {
-    this->findAlphabetCharacters();
-    this->isGraphCyclic = false;
-    this->orderedAlphabet.clear();
-    this->charPriorityGraph.clear();
-    for (auto c : this->alphabetCharacters) {
-        this->charPriorityGraph[c] = vector<bool>(NUM_OF_NATIVE_CHARS, false);
-    }
+	this->findAlphabetCharacters();
+	this->isGraphCyclic = false;
+	this->orderedAlphabet.clear();
+	this->charPriorityGraph.clear();
+	for (auto c : this->alphabetCharacters) {
+		this->charPriorityGraph[c] = vector<bool>(NUM_OF_NATIVE_CHARS, false);
+	}
+	this->inDegree.resize(NUM_OF_NATIVE_CHARS, 0);
 }
 
 void AlphabetFromDictionary::findAlphabetCharacters() {
-    this->alphabetCharacters.clear();
-    vector<bool> isCharInDictionary(NUM_OF_NATIVE_CHARS, false);
-    for (string &word : this->dictionary) {
-        for (char c : word) {
-            if (!isCharInDictionary[c]) {
-                isCharInDictionary[c] = true;
-                this->alphabetCharacters.push_back(c);
-            }
-        }
-    }
+	this->alphabetCharacters.clear();
+	vector<bool> isCharInDictionary(NUM_OF_NATIVE_CHARS, false);
+	for (auto word : this->dictionary) {

```

Comment:
I guess that when you pressed (ctrl+k+d) it turned "string &word" back to "auto word". It doesn't really bother me but it is important to @TijanaSonjic.. Could you change it back?